samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -0,0 +1,251 @@
+  }
+
+  /**
+   * Configures event listeners for various call-related events.
+   */
+  async setupEventListeners() {
+    const events = {
+      'active-speaker-change': this.handleActiveSpeakerChange.bind(this),
+      error: this.handleError.bind(this),
+      'joined-meeting': this.handleJoin.bind(this),
+      'participant-joined': this.handleJoin.bind(this),
+      'participant-left': this.handleJoin.bind(this),
+      'participant-updated': this.handleParticipantStateUpdate.bind(this),
+      'track-started': this.displayTrack.bind(this),
+      'track-stopped': this.destroyTrack.bind(this),
+    };
+
+    // Register the event handlers with the call object.
+    Object.entries(events).forEach(([event, handler]) => {
+      this.call.on(event, handler);
+    });
+  }
+
+  /**
+   * Handler for participant join events, updating participant count in the UI.
+   */
+  handleJoin() {
+    document.getElementById('participant-count').textContent = `Participants: ${
+      Object.keys(this.call.participants()).length
+    }`;
+    this.getTrackStates();
+  }
+
+  /**
+   * Handles errors emitted from the Daily call object.
+   * @param {Object} e - The error event object.
+   */
+  handleError(e) {
+    console.error('DAILY SENT AN ERROR!', e);
+    if (e.error?.details?.sourceError) {
+      console.log('Original Error', e.error?.details?.sourceError);
+    }
+  }
+
+  /**
+   * Decides whether to display video or play audio based on the track type.
+   * @param {Object} event - The track event object.
+   */
+  displayTrack(event) {
+    if (event.track.kind === 'video') {
+      this.displayVideo(event);
+    } else {
+      this.playAudio(event);
+    }
+  }
+
+  /**
+   * Adds a <video> element to the DOM and attaches the video track to it.
+   * @param {Object} event - The track event object containing the video track.
+   */
+  displayVideo(event) {
+    let videosDiv = document.getElementById('videos');
+    let videoEl = document.createElement('video');
+    videosDiv.appendChild(videoEl);
+    videoEl.style.width = '100%';
+    videoEl.srcObject = new MediaStream([event.track]);
+    videoEl.play();
+  }
+
+  /**
+   * Adds an <audio> element to the DOM and attaches the audio track to it.
+   * @param {Object} event - The track event object containing the audio track.
+   */
+  playAudio(event) {
+    if (event.participant.local) {
+      return; // Avoid playing local audio
+    }
+    let audioEl = document.createElement('audio'); // Create <audio> element
+    document.body.appendChild(audioEl); // Add <audio> element to DOM
+    audioEl.srcObject = new MediaStream([event.track]); // Attach audio track to <audio> element
+    audioEl.play(); // Play audio track
+  }
+
+  /**
+   * Removes the track's element (audio or video) from the DOM when it stops.
+   * @param {Object} event - The track event object.
+   */
+  destroyTrack(event) {
+    let tracks = document.querySelectorAll('video, audio');
+    tracks.forEach((el) => {
```

Comment:
I think we need to be sure to set `srcObject` to `null` to avoid leaving `WebMediaPlayer`s dangling indefinitely. E.g.: https://github.com/daily-demos/modern-wordfare/pull/22

>

Reply:
Good catch. Thank you 🙏